### ISSUE TITLE
Mismatch between sources in `DenoisedSource` and `OrientedSource`

### ISSUE THREAD

**Issue by janden** [ComputationalCryoEM/ASPIRE-Python#1020]
Elaborating on https://github.com/ComputationalCryoEM/ASPIRE-Python/pull/1014#discussion_r1337199921_.

The issue is that both `OrientedSource` and `DenoisedSource` are supplied both a source and an algorithm (orientation estimator `CLOrient3D` or denoiser `Denoiser`). The latter does not have to be applied to the source supplied to the source, which may not be the best interface and could introduce problems.

One natural option would be to allow mismatched sources, but to do something more sensible, i.e., use the denoiser provided to `DenoisedSource` and apply it to the images provided in the source. Similarly, `OrientedSource` could provide orientation estimates for the new source with respect to the orientations obtained in the `CLOrient3D` (this would require newish algorithms, however).

Will think a little more about this and post any ideas here.

**Comment by garrettwrong** [comment]
Thanks. There has been a related suggestion for `OrientedSource`.  From discussions with Amit/Yoel, they've expressed a more idealized usage of `CLOrient3D` on a single image source to look like running once globally a batch of say 500, then using that result to assign orientations subsequent batches.  Motivation being speed there.  This is essentially the same idea, or at least would be covered use case if the above was implemented.

**Comment by janden** [comment]
Right. This was what I was thinking of. Would probably be good to couple this with some check to see if the sources are the same and if so re-use the already estimated orientations (a more sophisticated alternative would compare images, or hashes of images, to see when they can be re-used).
